youmubot-osu/src/discord/display.rs

```diff
@@ -86,8 +86,7 @@ mod scores {
         #[async_trait]
         impl pagination::Paginate for Paginate {
             async fn render(&mut self, page: u8, ctx: &Context, msg: &mut Message) -> Result<bool> {
-                let data = ctx.data.read().await;
-                let env = data.get::<OsuEnv>().unwrap();
+                let env = ctx.data.read().await.get::<OsuEnv>().unwrap().clone();
                 let osu_client = &env.client;
                 let beatmaps = &env.beatmaps;
                 let beatmap_cache = &env.oppai;
```

Comment:
avoid aliasing like these, it's somewhat an anti-pattern to alias things just for the sake of aliasing, cuz it implies some effects happening and makes lifetime reasoning harder

Reply:
I repeated this anti-pattern a bit too much. I will fix some of them that I consider problematic.